source/extensions/filters/common/rbac/matchers/upstream_ip_port.cc

```diff
@@ -41,17 +43,30 @@ bool UpstreamIpPortMatcher::matches(const Network::Connection&,
           StreamInfo::UpstreamAddress::key());
 
   bool is_match = false;
-  if (range_.isInRange(*address_obj.address_)) {
-    ENVOY_LOG(debug, "UpstreamIpPort matcher for range: {} evaluated to: true", range_.asString());
-    is_match = true;
+  if (cidr_) {
+    const auto range = Network::Address::CidrRange::create(*cidr_);
```

Comment:
You can create the range in the constructor to avoid doing this on every match. Just make the range itself optional or a unique_ptr.